### PR TITLE
Add troi preference for auto spotify export

### PR DESCRIPTION
**Pull request by amCap1712**
Add troi preferences in general and a particular preference to allow users to configure whether they want to automatically export daily jams playlists to spotify.

### CODE REVIEW

File: listenbrainz/troi/troi_bot.py [hunk 1]
```diff
@@ -25,12 +27,18 @@ def run_post_recommendation_troi_bot():
 
 def run_daily_jams_troi_bot():
     """ Top level function called hourly to generate daily jams playlists for users """
-
     # Now generate daily jams (and other in the future) for users who follow troi bot
     users = get_users_for_daily_jams()
+    # TODO: Add endpoint to configure auto-export to spotify and do check scopes,
```
Comment (amCap1712) [inline listenbrainz/troi/troi_bot.py:32]: To discuss, what should be done when a user disconnects their spotify account? Should we clear the export to spotify preference or should this code be made more intelligent to handle that case? Maybe a mix?

Comment (amCap1712) [reply listenbrainz/troi/troi_bot.py:32]: Handling spotify submission failue gracefully will also need changes in troi.

Comment (mayhem) [inline listenbrainz/troi/troi_bot.py:32]: To be done in this PR or for a new ticket?

File: listenbrainz/troi/troi_bot.py [hunk 2]
```diff
@@ -69,26 +78,40 @@ def make_playlist_from_recommendations(user):
...
+        }
+
+    playlist = generate_playlist(DailyJamsPatch(), args)
+
     if len(playlist.playlists) > 0:
         url = current_app.config["SERVER_ROOT_URL"] + "/playlist/" + playlist.playlists[0].mbid
-        enter_timeline_notification(user, """Your daily-jams playlist has been updated. <a href="%s">Give it a listen!</a>.""" % url)
+        message = f"""Your daily-jams playlist has been updated. <a href="{url}">Give it a listen!</a>."""
+
+        if len(playlist.playlists[0].external_urls) > 0:
```
Comment (mayhem) [inline listenbrainz/troi/troi_bot.py:110]: Thats a really nice touch!

**Review by mayhem — APPROVED** [review]
All the non-JS bits look good!

File: listenbrainz/troi/troi_bot.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/troi/troi_bot.py:32]: Already done.

Comment (amCap1712) [reply listenbrainz/troi/troi_bot.py:32]: Discussed and settled on latter.

**Review by MonkeyDo — APPROVED** [review]
:+1: for UI side of things

**Pull request merged**